gpii/node_modules/testing/src/Mocks.js

```diff
@@ -453,65 +475,68 @@ fluid.defaults("gpii.test.integration.mockSettingsHandlerRegistry.android", {
     }
 });
 
-gpii.test.integration.deviceReporterAware.mockDeviceReporters = {};
+fluid.registerNamespace("gpii.test.integration.mockDeviceReporterImpls");
 
-gpii.test.integration.deviceReporterAware.mockDeviceReporters.findNameInList = function (expectInstalled, name) {
+gpii.test.integration.mockDeviceReporterImpls.findNameInList = function (expectInstalled, name) {
     return expectInstalled.some(function (el) { return el === name; });
 };
 
-gpii.test.integration.deviceReporterAware.mockDeviceReporters.registryKeyExists = function (expectInstalled, hKey, path, subPath) {
+gpii.test.integration.mockDeviceReporterImpls.registryKeyExists = function (expectInstalled, hKey, path, subPath) {
     return expectInstalled.some(function (el) {
         return el.hKey === hKey && el.path === path && el.subPath === subPath;
     });
 };
 
-gpii.test.integration.deviceReporterAware.mockDeviceReporters.wmiSettingSupported = function (expectInstalled, namespace, query) {
+gpii.test.integration.mockDeviceReporterImpls.wmiSettingSupported = function (expectInstalled, namespace, query) {
     return expectInstalled.some(function (el) {
         return el.namespace === namespace && el.query === query;
     });
 };
 
-gpii.test.integration.deviceReporterAware.populate = function (that) {
-    fluid.each(that.options.deviceReporters, function (options, key) {
+gpii.test.integration.mockDeviceReporters.populate = function (that) {
+    var backMocks = fluid.transform(that.options.mockDeviceReporters, function () {
+        return {
+            expectInstalled: []
+        };
+    });
+    var devReporters = fluid.extend(backMocks, that.options.deviceReporters);
+    fluid.each(devReporters, function (options, key) {
         var mock = that.options.mockDeviceReporters[key];
-
         if (!mock) {
             var registered = fluid.keys(that.options.mockDeviceReporters).join(", ");
             fluid.fail("Error in dynamic device reporter mock configuration - handler " + key +
                 " is not registered - registered handlers are " + registered);
         }
 
-        var expectInstalled = [];
-        fluid.each(that.options.deviceReporters[key].expectInstalled, function (item) {
-            expectInstalled.push(item);
-        });
-
         fluid.defaults(that.options.rootPath + "." + key, mock.defaults);
+        var mockFunc = fluid.getGlobalValue(mock.mockFunc);
 
-        var mockedFunction = gpii.test.integration.deviceReporterAware.mockDeviceReporters[mock.mockFunc].bind(undefined, expectInstalled);
+        var mockedFunction = mockFunc.bind(undefined, options.expectInstalled);
         fluid.setGlobalValue(that.options.rootPath + "." + key, mockedFunction);
     });
 };
 
-gpii.test.integration.deviceReporterAware.resolveName = function (that, name) {
+gpii.test.integration.mockDeviceReporters.depopulate = function (that) {
+    // mock deviceReporters are not components and so do not need destruction
```

Comment:
If it's genuinely a matter of component-ness or not, then as suggested before you could have a single function that uses `fluid.isComponent` instead of having both functions.

Reply:
Unfortunately these different categories of mocks don't actually share any common implementation so there's not great scope for reuse here in these 1-line functions. e.g. the settingsHandlers store their components in that.settingsHandlers and the others don't, etc.

Reply:
Yes, I guess I was reacting to the later clearing that both did, and assuming that the fluid.each call against that.settings would just be skipped for the mock device reporters.  It's such a small amount of material, there's not much point in reusing it.  However, the mockDeviceReporters function is now so slim that it could be replaced with an invoker that just calls fluid.setGlobalValue instead of the standalone function.